server/service/src/invoice/inbound_shipment/delete/validate.rs

```diff
@@ -12,43 +11,26 @@ pub fn validate(
     input: &DeleteInboundShipment,
     store_id: &str,
 ) -> Result<InvoiceRow, DeleteInboundShipmentError> {
-    let invoice = check_invoice_exists(&input.id, connection)?;
+    use DeleteInboundShipmentError::*;
 
-    check_store(&invoice, store_id)?;
-    check_invoice_type(&invoice, InvoiceRowType::InboundShipment)?;
-    check_invoice_is_editable(&invoice)?;
+    let invoice = check_invoice_exists_option(&input.id, connection)?.ok_or(InvoiceDoesNotExist)?;
+    if !check_store(&invoice, store_id) {
+        return Err(NotThisStoreInvoice);
+    }
+    if !check_invoice_type(&invoice, InvoiceRowType::InboundShipment) {
+        return Err(NotAnInboundShipment);
+    }
+    if !check_invoice_is_editable(&invoice) {
+        return Err(CannotEditFinalised);
+    }
 
     // check_invoice_is_empty(&input.id, connection)?; https://github.com/openmsupply/remote-server/issues/839
 
     Ok(invoice)
 }
 
-impl From<WrongInvoiceRowType> for DeleteInboundShipmentError {
-    fn from(_: WrongInvoiceRowType) -> Self {
-        DeleteInboundShipmentError::NotAnInboundShipment
-    }
-}
-
-impl From<InvoiceIsNotEditable> for DeleteInboundShipmentError {
-    fn from(_: InvoiceIsNotEditable) -> Self {
-        DeleteInboundShipmentError::CannotEditFinalised
-    }
-}
-
-impl From<InvoiceDoesNotExist> for DeleteInboundShipmentError {
-    fn from(_: InvoiceDoesNotExist) -> Self {
-        DeleteInboundShipmentError::InvoiceDoesNotExist
-    }
-}
-
 impl From<InvoiceLinesExist> for DeleteInboundShipmentError {
     fn from(error: InvoiceLinesExist) -> Self {
         DeleteInboundShipmentError::InvoiceLinesExists(error.0)
     }
 }
```

Comment:
is this one still needed?

Reply:
Well, you can delete invoices even if it has lines... so maybe not? 🤔

Reply:
sorry meant why was the From still needed? the other has been remove...

Reply:
Ah, cause it wasn't being used anywhere and wasn't sure if it was alright to remove now XD